ssl/ssl_mcnf.c

```diff
@@ -66,8 +66,10 @@ static int ssl_do_config(SSL *s, SSL_CTX *ctx, const char *name, int system)
         flags |= SSL_CONF_FLAG_CLIENT;
     SSL_CONF_CTX_set_flags(cctx, flags);
     prev_libctx = OSSL_LIB_CTX_set0_default(libctx);
+    err = 0;
     for (i = 0; i < cmd_count; i++) {
         char *cmdstr, *arg;
+        int rv;
 
         conf_ssl_get_cmd(cmds, i, &cmdstr, &arg);
         rv = SSL_CONF_cmd(cctx, cmdstr, arg);
```

Comment:
Theres some wierd interaction with SSL_CONF_cmd and the error checking below. SSL_CONF_cmd does some inconsistent error handling in the return 0 case:
1) if cmd == NULL it places an SSR_R_INVALID_NULL_CMD_NAME on the error queue and returns 0
2) if runcmd->cmd returns 0 in SSL_CONF_cmd it places SSL_R_BAD_VALUE on the error queue, iff cctx->flags & SSL_CONF_FLAG_SHOW_ERRORS is true

with the error handling below the error queue output will look somewhat odd:

* a NULL cmd will result in an error on the queue regardless of whether the SSL_CONF_FLAG_SHOW_ERRORS flag is set
* a zero return code from a valid conf cmd will show 2 SSL_R_BAD_VALUE entries on the queue if SSL_CONF_FLAG_SHOW_ERRORS is set, but still show 1 SSL_R_BAD_VALUE if SSL_CONF_FLAG_SHOW_ERRORS is unset.

Suggest that it might be better to move the ERR_raise_data be removed from line 77 below, and added to the cmd== NULL, value == NULL and rv == -2 if conditionals in SSL_CONF_cmd, gated by the SSL_CONF_FLAG_SHOW_ERRORS flag.  That consolidates  the error reporting, and just leaves the ++err to be handled here.

Reply:
Fixed, please look